### ISSUE TITLE
Elasticsearch registry stuck counters

### ISSUE THREAD

**Issue by ItayDev** [micrometer-metrics/micrometer#2469]
Hello, I tried to use the elastic registry (version 1.6.4) and it seems like it makes my counters have incorrect results.
Without installing this registry, when I ran a simple code which increments some counters and prints their results it worked fine. After installing this registry the counters seem to be stuck and not update their values correctly.
The documents are written to elasticsearch like they should but it seems the counters quickly stop increasing their values.
Other types of metrics (at least gauges from what I've seen) work correctly.
It would be great if you'll be able to help me resolve this issue, thanks in advance

**Comment by jonatan-ivanov** [comment]
@ItayDev Could you please prepare a minimal sample project that demonstrates the issue?

**Comment by ItayDev** [comment]
Sure, I'll do it soon (hopefully tomorrow) and will put a link to it here

**Comment by ItayDev** [comment]
Hey, so here is a sample repo I created: https://github.com/ItayDev/micrometer-elastic-sample
As you can see I use java 11, and simply increment and print a counter once every second.
Without the elasticsearch registry it increments and display the value correctly, after I added the dependency the 
counter doesn't increment like it should and usually get stuck pretty quickly. I also ran elasticsearch locally and checked
that it actually writes my metrics - and it does, only of course the counters won't increment so their value is not correct.
Hope this helps. Thanks

**Comment by shakuzen** [comment]
What are you expecting the output to be and what are you seeing it as? I think this may just be a difference in expectations.
When I am running it, the count appears frozen at 10, but this makes sense given the step interval is configured to be 10 seconds, and the scheduled task incrementing 1 happens every second. It is reporting a count of 10 over the previous interval (of 10 seconds).
See our documentation on [client-side rate aggregation](https://micrometer.io/docs/concepts#_client_side). Perhaps we could improve our documentation somehow to make clear this is the expected behavior in this case. Please feel free to share any suggestions you may have on this.
Does that clear things up?

**Comment by ItayDev** [comment]
Oh wow this is pretty awesome actually, I was expecting it to show the counter value itself and not the rate of the counter.

This is great, I somehow missed that it does that and thought I need to implement this behaviour myself.

And yes, it was a bit unclear to me that the elastic-output does it, generally the documentation I saw here https://micrometer.io/docs/registry/elastic was a bit lacking in configuration options and I needed to dig a bit around to see all the options such as `auto-create-index`.

Thank you very much!
